core/src/main/java/org/infinispan/factories/impl/ComponentRegistryTimeTracker.java

```diff
@@ -0,0 +1,236 @@
+      initialization.instantiated(timeService.wallClockTime());
+   }
+
+   @Override
+   public void wiring(String componentName) {
+      if (!isTimeServiceAvailable()) return;
+      Initialization initialization = componentInitialization(componentName);
+      initialization.wiring(timeService.wallClockTime());
+   }
+
+   @Override
+   public void wired(String componentName) {
+      if (!isTimeServiceAvailable()) {
+         if (componentName.equals(GlobalConfiguration.class.getName())) {
+            timeService = registry.getComponent(TimeService.class).wired();
+         }
+         return;
+      }
+      Initialization initialization = componentInitialization(componentName);
+      initialization.wired(timeService.wallClockTime());
+   }
+
+   @Override
+   public void starting(String componentName) {
+      if (!isTimeServiceAvailable()) return;
+      Initialization initialization = componentInitialization(componentName);
+      initialization.starting(timeService.wallClockTime());
+   }
+
+   @Override
+   public void started(String componentName) {
+      if (!isTimeServiceAvailable()) return;
+      Initialization initialization = componentInitialization(componentName);
+      initialization.started(timeService.wallClockTime());
+   }
+
+   @Override
+   public String dump() {
+      StringBuilder sb = new StringBuilder();
+      List<Initialization> c = components.values().stream()
+            .sorted(Comparator.reverseOrder())
+            .toList();
+      for (Initialization initialization : c) {
+         sb.append(initialization.name()).append(":").append(System.lineSeparator());
+         sb.append('\t').append("-> ").append("instantiating: ").append(initialization.instantiatingTime()).append(" ms").append(System.lineSeparator());
+         sb.append('\t').append("-> ").append(" instantiated: ").append(initialization.instantiatedTime()).append(" ms").append(System.lineSeparator());
+         sb.append('\t').append("-> ").append("       wiring: ").append(initialization.wiringTime()).append(" ms").append(System.lineSeparator());
+         sb.append('\t').append("-> ").append("        wired: ").append(initialization.wiredTime()).append(" ms").append(System.lineSeparator());
+         sb.append('\t').append("-> ").append("     starting: ").append(initialization.startingTime()).append(" ms").append(System.lineSeparator());
+         sb.append('\t').append("-> ").append("        total: ").append(initialization.totalTime()).append(" ms").append(System.lineSeparator());
+         sb.append(System.lineSeparator());
+      }
+      return sb.toString();
+   }
+
+   @Override
+   public void clear() {
+      components.clear();
+   }
+
+   private Initialization componentInitialization(String componentName) {
+      return components.computeIfAbsent(componentName, Initialization::new);
+   }
+
+   private static final class Initialization implements Comparable<Initialization> {
+      private static final long NOT_INITIALIZED = 0L;
```

Comment:
I wonder if this should be -1 so we can detect if a component was not tracked or a double update? Otherwise I can't tell the difference between really fast and not captured.